### PR TITLE
undefined rule: add explanation, add exceptions (closes #15)

### LINKED ISSUE #15: Checking whether variable is undefined
Not everyone is so happy to be able to assume that global `undefined` is untouched by user.
I suggest to change `undefined` checking to `typeof x == 'undefined'` since it's just safer.

### PR DESCRIPTION
**Pull request by ikokostya**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -421,21 +421,56 @@ The `eval` function should be avoided.
...
 ```
 
+**Exceptions:**
+
+  * `typeof` should be used if you need to support old browsers (like `IE8`) where `window.undefined` property is mutable.
+  * `typeof` should be used for detect undeclared global variables in **different** environments:
+
+  ```js
+  // It works in browser environment only.
+  if (window.feature !== undefined) {
```
Comment (ikokostya) [inline README.md]: MDN [recommends](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#Typeof_operator_and_undefined) (?) use checking property of global object:

> ...but the global scope is bound to the global object, so checking the existence of a variable in the global context can be done by checking the existence of a property on the global object (using the in operator, for instance).

Personally I prefer use
```js
if (typeof Int32Array !== 'function') {}
```
instead
```js
if (typeof window.Int32Array !== 'function') {}
```
And I don't see here difference with `undefined` checking. Should we use typeof with global variables?

Comment (ikokostya) [inline README.md]: There is my suggestion:

* If variable is declared and we want to check that is not `undefined` -> use equality with `undefined`
* If variable may not be declared -> use `typeof` or checking property of the global object.

**Comment by dodev** [comment]
:+1:

**Comment by dmikis** [comment]
LGTM

**Pull request merged**